rx-preferences/src/main/java/com/f2prateek/rx/preferences2/RealPreference.java

```diff
@@ -80,7 +79,9 @@
   }
 
   @Override public void delete() {
```

Comment:
(btw, the white-box test would look like this:)
```java
@Test public void foo() {
    final Preference<String> preference =
        new RealPreference<>(preferences, "foo", "", new RealPreference.Adapter<String>() {
          @Override public String get(@NonNull String key, @NonNull SharedPreferences preferences) {
            try {
              Thread.sleep(2000L);
            } catch (InterruptedException e) {
              throw new AssertionError(e);
            }
            String value = preferences.getString(key, null);
            if (value == null) {
              throw new AssertionError("Not called unless key is present.");
            }
            return value;
          }

          @Override public void set(@NonNull String key, @NonNull String value,
              @NonNull SharedPreferences.Editor editor) {
            editor.putString(key, value);
          }
        }, Observable.<String>empty());
    preference.set("hello");
    new Thread(new Runnable() {
      @Override public void run() {
        try {
          Thread.sleep(1000L);
        } catch (InterruptedException e) {
          throw new AssertionError(e);
        }
        preference.delete();
      }
    }).start();
    preference.get();
  }
```
(that would fail without locking.)